erpnext/controllers/accounts_controller.py

```diff
@@ -1290,6 +1294,11 @@ def validate_quantity(child_item, d):
 				child_item.conversion_factor = 1
 			else:
 				child_item.conversion_factor = flt(d.get('conversion_factor'))
+		
+		if d.get("uom"):
+			child_item.uom = d.get("uom")
+			conversion_factor = flt(get_conversion_factor(child_item.item_code, child_item.uom).get("conversion_factor"))
+			child_item.conversion_factor = flt(d.get('conversion_factor')) or conversion_factor
```

Comment:
which conversion factor should have precedence?

Reply:
@nextchamp-saqib the one set in trans_items(set by user) should have precedence as conversion factor was added for that very reason.
Although on the ux end if you want to, we could fetch and set conversion factor **on changing uom only**, after that if they want to set a custom conversion factor they can. 
Because we know of only 1 use case where conversion factor changes all the time. 90% of the time it will change with uom mostly, so it will save such users extra data entry.
Just a suggestion

Reply:
Just added the conversion factor fetch on UOM change